_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+}
+var unHighlight = function(id) {
+  $("#r-" + id).attr("stroke", "none")
+  $("#sc-" + id).css("background-color", "white")
+}
+
+$("svg").find("g#React > rect").hover(function(){
+    var newID = this.id.replace(/^r-/, "")
+    highlight(newID)
+}, function () {
+    var newID = this.id.replace(/^r-/, "")
+    unHighlight(newID)
+});
+
+$(".component").hover(function(){
+    var newID = this.id.replace(/^sc-/, "")
+    highlight(newID)
+}, function () {
+    var newID = this.id.replace(/^sc-/, "")
+    unHighlight(newID)
+});
+</script>
+</article>
+<article class="post" style="margin-top: 620px">
+
+This kind of tree structure should feel quite similar to the view tree that you see inside a tool like Reveal, or inside the Xcode visual inspector.
+
+This is a *simplified* version of what that code for the component above looks like, you can see the [full file here](https://github.com/artsy/emission/blob/c1ccd63ec9f70c2817186cb555bfd874c375912a/src/lib/components/consignments/components/artist_search_results.tsx).
+
+{% raw %}
+<!-- The {{ and }} get eaten by mustache -->
+```js
+// Import React, and native components from React Native
+import * as React from "react"
+import { ScrollView, Text, Image, View } from "react-native"
+
+// Re-use our existing search TextInput component
+import TextInput from "./text_input"
+
+// Exports a React component called Search Results from this file
+export default class SearchResults extends React.Component {
+
+  // The tree of components that this component represents
+  render() {
+    // This is JSX code, JSX is a source-code transformer that converts code from 
+    // HTML-like brackets into a specific method call. E.g  `<Text font="Garamond" />`
+    // turns into; `React.createElement('Text', {font: 'Garamond'}, null)`
+    return (
+      <View>
+        <TextInput text={{ value: props.query }} searching={props.searching} />
+        <ScrollView>
+          {props.results.map(rowForResult)}
+        </ScrollView>
+      </View>
+    )
+  }
+  // Returns a single component for a row
+  rowForResult(result) {
+    return (
+      <Result>
+        <Image source={{ uri: result.url }} />
+        <Text>{result.name}</Text>
+      </Result>
+    )
+  }
+}
+```
+{% endraw %}
+
+> You're looking at a subclass of `React.Component` with two functions, `render` and `rowForResult`. `render` is the key function for defining your tree.
+
+Instead of MVC, React uses composition of components to handle complexity - this should feel quite similar to iOS development. The screen of an iOS app is typically made up of `UIView`s, and `UIViewController`s which exist as 2 interlinked trees of hierarchy. A `UIViewController` itself doesn't have a visual representation, but exists to manipulate data, handle actions and the view structure for `UIViews` who do. 
+
+By merging the responsibilities of a `UIView` and `UIViewController` into a Component, there is a consistent way to work with all aspects of your app. 
```

Comment:
Could be worth alluding to other approaches that do this, like MVVM. Or not. Up to you!